eclair-core/src/test/scala/fr/acinq/eclair/router/ChannelRangeQueriesSpec.scala

```diff
@@ -127,4 +131,226 @@ class ChannelRangeQueriesSpec extends FunSuite {
+      val firstBlockNum = 900
+      val numberOfBlocks = 200
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 2)
+      assert(chunks == ShortChannelIdsChunk(firstBlockNum, numberOfBlocks, ids) :: Nil)
+    }
+
+    // all ids in different blocks, chunk size == 2
+    {
+      val ids = List(id(1000), id(1001), id(1002), id(1003), id(1004), id(1005))
+      val firstBlockNum = 900
+      val numberOfBlocks = 200
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 2)
+      assert(chunks == List(
+        ShortChannelIdsChunk(firstBlockNum, 100 + 2, List(ids(0), ids(1))),
+        ShortChannelIdsChunk(1002, 2, List(ids(2), ids(3))),
+        ShortChannelIdsChunk(1004, numberOfBlocks - 1004 + firstBlockNum, List(ids(4), ids(5)))
+      ))
+    }
+
+    // all ids in different blocks, chunk size == 2, first id outside of range
+    {
+      val ids = List(id(1000), id(1001), id(1002), id(1003), id(1004), id(1005))
+      val firstBlockNum = 1001
+      val numberOfBlocks = 200
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 2)
+      assert(chunks == List(
+        ShortChannelIdsChunk(firstBlockNum, 2, List(ids(1), ids(2))),
+        ShortChannelIdsChunk(1003, 2, List(ids(3), ids(4))),
+        ShortChannelIdsChunk(1005, numberOfBlocks - 1005 + firstBlockNum, List(ids(5)))
+      ))
+    }
+
+    // all ids in different blocks, chunk size == 2, last id outside of range
+    {
+      val ids = List(id(1000), id(1001), id(1002), id(1003), id(1004), id(1005))
+      val firstBlockNum = 900
+      val numberOfBlocks = 105
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 2)
+      assert(chunks == List(
+        ShortChannelIdsChunk(firstBlockNum, 100 + 2, List(ids(0), ids(1))),
+        ShortChannelIdsChunk(1002, 2, List(ids(2), ids(3))),
+        ShortChannelIdsChunk(1004, numberOfBlocks - 1004 + firstBlockNum, List(ids(4)))
+      ))
+   }
+
+    // all ids in different blocks, chunk size == 2, first and last id outside of range
+    {
+      val ids = List(id(1000), id(1001), id(1002), id(1003), id(1004), id(1005))
+      val firstBlockNum = 1001
+      val numberOfBlocks = 4
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 2)
+      assert(chunks == List(
+        ShortChannelIdsChunk(firstBlockNum, 2, List(ids(1), ids(2))),
+        ShortChannelIdsChunk(1003, 2, List(ids(3), ids(4)))
+      ))
+    }
+
+    // all ids in the same block
+    {
+      val ids = makeShortChannelIds(1000, 100)
+      val firstBlockNum = 900
+      val numberOfBlocks = 200
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 10)
+      assert(chunks == ShortChannelIdsChunk(firstBlockNum, numberOfBlocks, ids) :: Nil)
+    }
```

Comment:
I tried to add the following test and it doesn't pass. I think we should change the code to make it pass :)

```scala
    // don't start a new block in same chunk if it overflows chunk size
    {
      val ids = List(id(1000), id(1000, 1), id(1001), id(1001, 1))
      val firstBlockNum = 900
      val numberOfBlocks = 200
      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, firstBlockNum, numberOfBlocks, 3)
      assert(chunks == ShortChannelIdsChunk(firstBlockNum, 100 + 1, List(ids(0), ids(1))) :: ShortChannelIdsChunk(1001, 1, List(ids(2), ids(3))) :: Nil)
    }
```

When we decide to start adding a new block to the current chunk, we should only do that if the whole block can fit in the chunk. Otherwise we should take the safer choice of starting a new chunk.
If we don't do that it's likely we'll run more often in the code that does the random split to avoid the 3200 elements limit which I think we should try to avoid (it's not really a problem to send a bit more chunks imo).

Reply:
Yes the current "merge" algo is simplistic (but also very efficient). Being smarter would require to iterate twice and computing a `height -> count` map.
I propose instead to change the default `channel-range-chunk-size` to 1500 instead of 2500, it should reduces the probability of merging 2 chunks and ending up over the limit

Reply:
SGTM